pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -90,9 +91,11 @@
 
   // Pre-calculate the aggregation functions and columns for the query so that it can be shared across all the segments
   private AggregationFunction[] _aggregationFunctions;
-  private List<Pair<AggregationFunction, FilterContext>> _filteredAggregationFunctions;
-  // TODO: Use Pair<FunctionContext, FilterContext> as key to support filtered aggregations in order-by and post
-  //       aggregation
+
+  private List<Pair<FilterContext, AggregationFunction>> _aggregationFunctionsWithMetadata;
```

Comment:
Let's rename it to `_filteredAggregations` (it has nothing to do with metadata)? Also suggest putting `AggregationFunction` as the first argument of the pair to be consistent with other fields